packages/nimble-components/src/wafer-map/modules/wafer-map-validator.ts

```diff
@@ -0,0 +1,43 @@
+import type { WaferMap } from '..';
+import type { WaferMapValidity } from '../types';
+
+/**
+ * Helper class for the nimble-table to validate that the table's configuration
+ * is valid and report which aspects of the configuration are valid or invalid.
+ */
+export class WaferMapValidator {
+    private invalidGridDimensions = false;
+
+    public constructor(private readonly wafermap: WaferMap) {}
+    public getValidity(): WaferMapValidity {
+        return {
+            invalidGridDimensions: this.invalidGridDimensions
+        };
+    }
+
+    public isValid(): boolean {
+        return Object.values(this.getValidity()).every(x => x === false);
+    }
+
+    public validateGridDimensions(): boolean {
+        this.invalidGridDimensions = false;
+        if (
+            this.wafermap.gridMinX === undefined
```

Comment:
A slightly stronger check would be checking that they are all `typeof` 'number' here and below.

There are known issues where removing an attribute results in `null` instead of `undefined` (i.e. `waferMap.removeAttribute('grid-min-x')`) even though the types don't show that. A stricter check using typeof number would prevent that issue.